Code/Tools/FBuild/FBuildCore/Graph/ObjectListNode.cpp

```diff
@@ -353,7 +353,20 @@ bool ObjectListNode::CreateDynamicObjectNode( Node * inputFile, const DirectoryL
             // ... use everything after that
             subPath.Assign(fileName.Get() + baseDir.GetLength(), lastSlash ); // includes last slash
         }
-    }
+	}
+	else
+	{
+		// if source comes from a specific list of files, use path relative to file base directory
+		if ( inputFile->GetType() == Node::FILE_NODE )
+		{
+			FileNode * fileNode = inputFile->CastTo< FileNode >();
+			const AString & baseDir = fileNode->GetBaseDirectory();
+			if (baseDir.GetLength() > 0)
```

Comment:
This needs to check if fileName begins with the baseDir (using PathUtils::PathBeginsWith) otherwise you could end up with a corrupt filename, or worse, a crash (if baseDir was longer than fileName)